helix-core/src/syntax.rs

```diff
@@ -73,16 +73,37 @@ pub struct IndentQuery {
     pub outdent: HashSet<String>,
 }
 
-fn read_query(language: &str, filename: &str) -> String {
-    static INHERITS_REGEX: Lazy<Regex> =
-        Lazy::new(|| Regex::new(r";+\s*inherits\s*:?\s*([a-z_,()]+)\s*").unwrap());
-
+#[cfg(not(feature = "embed_runtime"))]
+fn load_runtime_file(language: &str, filename: &str) -> Result<String, std::io::Error> {
     let root = crate::runtime_dir();
-    // let root = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    let path = root.join("queries").join(language).join(filename);
+    std::fs::read_to_string(&path)
+}
+
+#[cfg(feature = "embed_runtime")]
+use rust_embed::RustEmbed;
+
+#[cfg(feature = "embed_runtime")]
+#[derive(RustEmbed)]
+#[folder = "../runtime/"]
+struct Runtime;
 
+#[cfg(feature = "embed_runtime")]
+fn load_runtime_file(language: &str, filename: &str) -> Result<String, Box<dyn std::error::Error>> {
+    let root = PathBuf::new();
     let path = root.join("queries").join(language).join(filename);
 
-    let query = std::fs::read_to_string(&path).unwrap_or_default();
+    let query_bytes = Runtime::get(&path.as_path().display().to_string()).unwrap_or_default();
+    std::str::from_utf8(query_bytes.as_ref())
+        .map(|s| s.to_string())
+        .map_err(|err| err.into())
```

Comment:
Do we want this to error? I think it should not happen though, we could just put `expect`, since we embed this I don't think it should fail.

Reply:
I figured since the non-embedded version returns an error that I would continue to do so to preserve the API, you are right that this should not happen and if for some reason it did fail the user wouldn't really have any recourse to respond to it. I am open to either, it does feel odd to unwrap in a function that has to return a result anyways.

+1 to writing a test, I just added a basic smoke test to make sure we can read any runtime data from either the rust_embed or the default codepaths.

Reply:
No, using an `Option` is not a good idea because it hides the error. Imagine if we unwrap and it gets a `NoneError`.

Reply:
Hmm yeah you're right, I'll undo that. I do think we probably want a result type for both branches so when I get some free time I'll look into how best to do it. I would normally reach for `anyhow` here but I might avoid pulling in a dependency just for this situation.

Reply:
OK ended up getting it working, it wasn't too bad. Now both versions of the `load_runtime_file` function will return an error if the file requested does not exist.